---
title: "Wild Damsel Fish Analysis"
author: "Christopher Kavazos"
date: "15 June 2020"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

This document details the analysis of the sequence data output from QIIME2 for the natural Damsel fish populations. All images/graphs are produce in this documentas pdfs.

# Load packages

```{r echo=TRUE, include=FALSE}
library(phyloseq)
library(ggplot2)
library(data.table)
library(vegan)
library(Biostrings)
library(plyr)
library(DESeq2)
library(dplyr)
library(egg)
library(mvabund)
library(XLConnect)
library(gplots)
library(VennDiagram)
library(gridBase)
library(grid)
library(gridExtra)
library(cowplot)
library(ape)
library(phangorn)

source("source_functions.R")

set.seed(100)
```

# Import Data

This section imports the data as .biom and .fasta files in the workspace as phyloseq objects.

```{r echo=TRUE, include=TRUE, tidy=TRUE, warning=FALSE}
load("raw_data.RData")
seqs <- refseq(data)
```

# Pre-processing of data

## Remove unwanted sequences

We exclude all unkown sequences that probably belong to the host. Chloroplast and cyanobacterial sequences are also removed as they belong to the food items in the host's guts at the time of sampling.

```{r echo=TRUE, include=TRUE, tidy=TRUE}
ps <- subset_taxa(data, (Class!="Chloroplast") | is.na(Class))
ps <- subset_taxa(ps, (Family!="mitochondria") | is.na(Family))
ps <- subset_taxa(ps, (Phylum!="Cyanobacteria") | is.na(Phylum))
ps <- subset_taxa(ps, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
ps
```

# Basic Stats

The following step provide basic information on the dataset.

## Total Reads

```{r echo=TRUE, include=TRUE, tidy=TRUE}
sdt <- data.table(as(sample_data(ps), "data.frame"),
TotalReads=sample_sums(ps), keep.rownames = TRUE)
colSums(sdt[,"TotalReads"])
```

## Number of Taxa

```{r echo=TRUE, include=TRUE, tidy=TRUE}
ntaxa(ps)
```

## Most abundant Phyla

```{r echo=TRUE, include=TRUE, tidy=TRUE}
table(tax_table(ps)[, "Phylum"], exclude = NULL)

sort(taxa_sums(ps), decreasing =TRUE)[1:30]

ps20 <- subset_taxa(ps, rownames(tax_table(ps)) %in% names(sort(taxa_sums(ps), decreasing=TRUE)[1:30]))
```

# Build phyloseq object of each individual
Here we merge the four samples for each individual so that we have data on all the sequences detected within each individual.

```{r echo=TRUE, include=TRUE, tidy=TRUE}

psI <- merge_samples(ps, "Individual")
sample_data(psI)$Individual <- factor(sample_names(psI))
psI <- merge_phyloseq(psI, seqs)
otu_table(psI) <- t(otu_table(psI))
psI

# Build new sample_data
df <- as.data.frame(lapply(sample_data(psI),function (y) if(class(y)!="num" ) as.factor(y) else
  y),stringsAsFactors=T)
row.names(df) <- sample_names(psI)
sample_data(psI) <- sample_data(df)
metadt <- as.data.frame(sample_data(psI))
FishSpec <- revalue(get_variable(metadt, "FishSpecies"), c("6"="polyacanthus", "8"="sexfasciatus", "10"="whitleyi",
                                                           "2"="atripectoralis", "5"="perspicillatus",
                                                           "7"="pseudochrysopoecilus", "3"="moluccensis",
                                                           "9"="wardi", "1"="apicalis", "4"="nigricans"))
Treat <- revalue(get_variable(metadt, "Treatment"), c("1"="Planktivorous", "2"="Territorial"))
metadt$FishSpecies <- FishSpec
metadt$Treatment <- Treat
sample_data(psI) <- metadt
```

# Alpha Diversity

##Rarefy to even sampling depth

Before assessing alpha richness, we rarefy the data to a common sammple size - here the minimum sample size.

```{r echo=TRUE, include=TRUE, tidy=TRUE}
psIrar <- rarefy_even_depth(psI, sample.size = min(sample_sums(psI)))
sample_sums(psIrar)
alphI <- estimate_richness(psIrar)
alphI["Species"] <- metadt[,"FishSpecies"]

alpha.sp <- NULL
for(i in levels(alphI$Species)){
  tmp<-data.frame(mean(subset(alphI, Species==i)$Observed), sd(subset(alphI, Species==i)$Observed))
  if(is.null(alpha.sp )){alpha.sp <-tmp} else { alpha.sp <-rbind(alpha.sp ,tmp)} 
}
row.names(alpha.sp) <- levels(alphI$Species)
colnames(alpha.sp) <- c("mean", "sd")
alpha.sp
```

## Construct plots

```{r echo=TRUE, include=TRUE, tidy=TRUE}
alphI$Species <- factor(alphI$Species,
                        levels = c("polyacanthus", "sexfasciatus",  "whitleyi", "atripectoralis",
                                   "moluccensis", "perspicillatus", "pseudochrysopoecilus", "wardi", 
                                   "apicalis", "nigricans"))

levels(alphI$Species)[levels(alphI$Species)=="polyacanthus"] <- "A. polyacanthus"
levels(alphI$Species)[levels(alphI$Species)=="sexfasciatus"] <- "A. sexfasciatus"
levels(alphI$Species)[levels(alphI$Species)=="whitleyi"] <- "A. whitleyi"
levels(alphI$Species)[levels(alphI$Species)=="atripectoralis"] <- "C. atripectoralis"
levels(alphI$Species)[levels(alphI$Species)=="moluccensis"] <- "P. moluccensis"
levels(alphI$Species)[levels(alphI$Species)=="perspicillatus"] <- "D. perspicillatus"
levels(alphI$Species)[levels(alphI$Species)=="pseudochrysopoecilus"] <- "D. pseudochrysopoecilus"
levels(alphI$Species)[levels(alphI$Species)=="wardi"] <- "P. wardi"
levels(alphI$Species)[levels(alphI$Species)=="apicalis"] <- "S. apicalis"
levels(alphI$Species)[levels(alphI$Species)=="nigricans"] <- "S. nigricans"
                       
obs <- ggplot(alphI, aes(Species,Observed)) +
  geom_boxplot(aes(fill = Species)) +
  ylab("Observed ASVs") +
  theme_classic() +
  stat_summary(fun=mean, geom="point", shape=5, size=4) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + theme(legend.position="none") +
  scale_fill_manual(values = c("A. polyacanthus" = "red",
                               "A. sexfasciatus" = "red",
                               "A. whitleyi" = "red",
                               "C. atripectoralis" = "red",
                               "P. moluccensis" = "red",
                               "D. perspicillatus" = "green",
                               "D. pseudochrysopoecilus" = "green",
                               "P. wardi" = "green",
                               "S. apicalis" = "green",
                               "S. nigricans" = "green"))

shan <- ggplot(alphI, aes(Species,Shannon)) +
  geom_boxplot(aes(fill=Species)) +
  ylab("Shannon Index") +
  xlab(" ") + theme_classic() +
  stat_summary(fun=mean, geom="point", shape=5, size=4) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + theme(legend.position="none") +
  scale_fill_manual(values = c("A. polyacanthus" = "red",
                               "A. sexfasciatus" = "red",
                               "A. whitleyi" = "red",
                               "C. atripectoralis" = "red",
                               "P. moluccensis" = "red",
                               "D. perspicillatus" = "green",
                               "D. pseudochrysopoecilus" = "green",
                               "P. wardi" = "green",
                               "S. apicalis" = "green",
                               "S. nigricans" = "green"))

Pielou = (alphI$Shannon/log(alphI$Observed))
alphI["Pielou"] = Pielou
piel <- ggplot(alphI, aes(Species,Pielou)) +
  geom_boxplot(aes(fill=Species)) +
  ylab("Pielou Evenness") +
  xlab(" ") + theme_classic() +
  stat_summary(fun=mean, geom="point", shape=5, size=4) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + theme(legend.position="none")+
  scale_fill_manual(values = c("A. polyacanthus" = "red",
                               "A. sexfasciatus" = "red",
                               "A. whitleyi" = "red",
                               "C. atripectoralis" = "red",
                               "P. moluccensis" = "red",
                               "D. perspicillatus" = "green",
                               "D. pseudochrysopoecilus" = "green",
                               "P. wardi" = "green",
                               "S. apicalis" = "green",
                               "S. nigricans" = "green"))

multiplot(obs,shan,piel, cols = 3)

# Save plot as a .pdf
ggsave("a_diveristy_plots.pdf", multiplot(obs,shan,piel, cols = 3), width=297, height=210, units="mm")
```

## Top 20 most abundant taxa

Here we extract the 20 most abundant ASVs and save as a phyloseq object 'ps20'

```{r echo=TRUE, include=TRUE, tidy=TRUE}

top20otus = names(sort(taxa_sums(ps), TRUE)[1:20])
ps20 <- subset(otu_table(psI), rownames(otu_table(psI)) %in% top20otus)
ps20 <- merge_phyloseq(ps20, tax_table(psI), sample_data(psI), phy_tree(psI),
                       refseq(psI))
sort(rowSums(otu_table(ps20)))
#tax_table(ps20)
#otu_table(ps20)
```

## Barplot 

Here we produce a barplot of the 20 most abundant ASV's relative abundances

```{r echo=TRUE, include=TRUE, tidy=TRUE}
top20otus = names(sort(taxa_sums(ps20), TRUE)[1:20])
taxtab20 = cbind(tax_table(ps20), family20 = NA)
taxtab20[top20otus, "family20"] <- as(tax_table(ps20)[top20otus, "Family"], 
                                      "character")
tax_table(ps20) <- tax_table(taxtab20)
ps20m = merge_samples(ps20, "FishSpecies")
sample_data(ps20m)$FishSpecies <- levels(sample_data(ps20)$FishSpecies)
ps20m = transform_sample_counts(ps20m, function(x) 100 * x/sum(x))
ps20m20 = prune_taxa(top20otus, ps20m)

barplot <- plot_bar(ps20m20, "FishSpecies", fill = "Class", title = NULL) + 
  theme_bw() + coord_flip() +
  labs(x=NULL, y=("Percentage of Sequences")) + ylim(0, 65) +
  theme(axis.line=element_line(),
        axis.line.y = element_line(),
        panel.background= element_blank(),
        panel.border = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank()) +
  guides(fill=guide_legend(title="Bacteria Families"))

barplot$data$FishSpecies <- as.character(barplot$data$FishSpecies)
barplot$data$FishSpecies <- factor(x = barplot$data$FishSpecies, 
                                   levels = 
                            unique(barplot$data$FishSpecies[order(as.factor(barplot$data$Treatment))]))

specieslabs <- c("A. sexfasciatus","A. polyacanthus","C. atripectoralis","P. moluccensis",
                 "A. whitleyi", "D. perspicillatus","S. nigricans","P. wardi","D. pseudochrysopoecilus",
                 "S. apicalis")

barplot <- barplot + scale_x_discrete(labels=specieslabs) + 
  theme(axis.text.y = element_text(face="italic"))
```
## Circular plot

Build circular plot of total pooled samples to show the relative abundances of each of the 20 most abundant ASVs

```{r, echo=TRUE, tidy=TRUE}

cir.dat <- data.frame(cbind(row.names(data.frame(tax_table(ps20m20))), as.character(data.frame(tax_table(ps20m20))[,2]),
                 as.character(data.frame(tax_table(ps20m20))[,5]), 
                 as.numeric(colSums(data.frame(otu_table(ps20m20))))))
colnames(cir.dat) <- c("ASV", "Phylum", "Family", "Abundance")
cir.dat$ASV <- NULL
cir.dat$Abundance <- as.integer(cir.dat$Abundance)

# Set a number of 'empty bar' to add at the end of each group
empty_bar=3
to_add = data.frame( matrix(NA, empty_bar*nlevels(cir.dat$Phylum), ncol(cir.dat)) )
colnames(to_add) = colnames(cir.dat)
to_add$Phylum=rep(levels(cir.dat$Phylum), each=empty_bar)
cir.dat=rbind(cir.dat, to_add)
cir.dat=cir.dat %>% arrange(Phylum)
cir.dat$id=seq(1, nrow(cir.dat))

# Get the name and the y position of each label
label_data=cir.dat
number_of_bar=nrow(label_data)
# I substract 0.5 because the letter must have the angle of the center of the bars. 
# Not extreme right(1) or extreme left (0)
angle= 90 - 360 * (label_data$id-0.5) /number_of_bar     
label_data$hjust<-ifelse( angle < -90, 1, 0)
label_data$angle<-ifelse(angle < -90, angle+180, angle)

# prepare a data frame for base lines
base_data=cir.dat %>% 
  group_by(Phylum) %>% 
  summarize(start=min(id), end=max(id) - empty_bar) %>% 
  rowwise() %>% 
  mutate(title=mean(c(start, end)))

# prepare a data frame for grid (scales)
grid_data = base_data
grid_data$end = grid_data$end[ c( nrow(grid_data), 1:nrow(grid_data)-1)] + 1
grid_data$start = grid_data$start - 1
grid_data=grid_data[-1,]

# Make the plot
cir.p = ggplot(cir.dat, aes(x=as.factor(id), y=Abundance, fill=Phylum)) +
  # Note that id is a factor. If x is numeric, there is some space between the first bar
    geom_bar(aes(x=as.factor(id), y=Abundance, fill=Phylum), stat="identity", alpha=0.5) +
    # Add a val=100/75/50/25 lines. I do it at the beginning to make sur barplots are OVER it.
  geom_segment(data=grid_data, aes(x = end, y = 20, xend = start, yend = 20), 
               colour = "grey", alpha=1, size=0.3 , inherit.aes = FALSE ) +
  geom_segment(data=grid_data, aes(x = end, y = 15, xend = start, yend = 15), 
               colour = "grey", alpha=1, size=0.3 , inherit.aes = FALSE ) +
  geom_segment(data=grid_data, aes(x = end, y = 10, xend = start, yend = 10), 
               colour = "grey", alpha=1, size=0.3 , inherit.aes = FALSE ) +
  geom_segment(data=grid_data, aes(x = end, y = 5, xend = start, yend = 5), 
               colour = "grey", alpha=1, size=0.3 , inherit.aes = FALSE ) +
  
  # Add text showing the value of each 100/75/50/25 lines
  annotate("text", x = rep(max(cir.dat$id),2), y = c(10, 20), 
           label = c("10", "20") , color="grey", size=3 , 
           angle=0, fontface="bold", hjust=1) +
  
  geom_bar(aes(x=as.factor(id), y=Abundance, fill=Phylum), stat="identity", alpha=0.5) +
  ylim(-100,80) +
  theme_minimal() +
  theme(
    legend.position = "none",
    axis.text = element_blank(),
    axis.title = element_blank(),
    panel.grid = element_blank(),
    plot.margin = unit(rep(-1,4), "cm") 
  ) +
  coord_polar() + 
  geom_text(data=label_data, aes(x=id, y=Abundance+10, label=Family, 
                                 hjust=hjust), color="black", fontface="bold",
            alpha=0.6, size=3, angle= label_data$angle, inherit.aes = FALSE ) +
  
  # Add base line information. Note hjust must have same number of values as groups
  geom_segment(data=base_data, aes(x = start, y = -5, xend = end, yend = -5), 
               colour = "black", alpha=0.8, size=0.6 , inherit.aes = FALSE )  +
  geom_text(data=base_data, aes(x = title, y = -18, label=Phylum), 
            hjust=c(1,0,0), colour = "black", alpha=0.8, size=4, 
            fontface="bold", inherit.aes = FALSE)
```

##  Build Figure 1

Produce a figure with the circular bar plot and fish species breakdown together. Additional format required in an editing program

```{r, echo=TRUE, tidy=TRUE}
ggarrange(cir.p,barplot, ncol=2, nrow=1)

ggsave(plot=ggarrange(cir.p,barplot, ncol=2, nrow=1),
       file = "Figure1.pdf", 
      width=396, height=280, units="mm",
       useDingbats=FALSE)
```

# CORE Microbiome

This analysis assess the different levels of core microbiomes follwoing the methodology of Gumiere et al. (2018).

```{r echo=TRUE, include=TRUE, tidy=TRUE}
psI2 <- merge_samples(ps, "Individual")
sample_data(psI2) <- metadt
psI2rar <- rarefy_even_depth(psI2, sample.size = min(sample_sums(psI2)), rngseed =100)

ps.otu2 <- data.frame(t(otu_table(psI2rar)))

core <- core.fit(ps.otu2)

# Conventional figure
#dev.off()
plot.new()
plot(freq~log(p),pch=18,col=ifelse(freq>=0.8,"blue",
                                   ifelse(freq>=0.7,"blue",
                                          ifelse(freq>=0.6,"blue",
                                                 ifelse(freq>=0.5,"blue",
                                                        ifelse(freq>=0.4,"blue",
                                                               ifelse(freq>=0.3,"blue",
                                                                      ifelse(freq>=0.3,"blue","gray")))))))
     ,ylab="Occurence frequency",las=1,xlab="log(Mean Relative Abundance)",core$otu_matrix,ylim=c(0,1))

lines(y=c(1,1),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.9,0.9),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.8,0.8),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.7,0.7),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.6,0.6),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.5,0.5),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.4,0.4),x=c(-9,0),lty=2,col="gray")
lines(y=c(0.3,0.3),x=c(-9,0),lty=2,col="gray")

text(-9,1.015,"100%",col="gray40",cex=0.8)
text(-9,0.915,"90%",col="gray40",cex=0.8)
text(-9,0.815,"80%",col="gray40",cex=0.8)
text(-9,0.715,"70%",col="gray40",cex=0.8)
text(-9,0.615,"60%",col="gray40",cex=0.8)
text(-9,0.515,"50%",col="gray40",cex=0.8)
text(-9,0.415,"40%",col="gray40",cex=0.8)
text(-9,0.315,"30%",col="gray40",cex=0.8)

text(-10.2,0.6,"core\ncommunity\n(thresholds)",col="blue")
text(-10.2,0.2,"variable\ncommunity",col="gray40")

# Save Core80 dataset
core.30 <- subset_taxa(psI2rar, rownames(tax_table(psI2rar)) %in% rownames(core$core_30))
tax_table(core.30)
```

## Prevalence
Here we calculate how many samples the ASVs ovvured in and their relative mean abundance using the rarefied data

```{r echo=TRUE, include=TRUE, tidy=TRUE}
prevelancedf = apply(X = otu_table(psI2rar),
                 MARGIN = 2,
                 FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevelancedf = data.frame(Prevalence = prevelancedf,
                      TotalAbundance = taxa_sums(psI2rar),
                      tax_table(psI2rar))

prevelancedf[order(-prevelancedf$Prevalence),][1:13,]

# Calculate total unmber of reads in rarefied data set

colSums(data.table(as(sample_data(psI2rar), "data.frame"),
TotalReads=sample_sums(psI2rar), keep.rownames = TRUE)[,"TotalReads"])

```

## Venn diagrams

Here we produce three sets of Venn diagrams. One for all the fish species, one for the different feeding behaviours and one for the different locations along the gut.

### Different host species

This section builds a new phyloseq object for each of the host fish species

```{r echo=TRUE, include=TRUE, tidy=TRUE}
Ind.lst <- list()
for(i in levels(data.frame(sample_data(psI2rar))$Individual)){
  tmp <- subset_samples(psI2rar, Individual==i)
  Ind.lst[[length(Ind.lst)+1]] = tmp
}

Ind.lst <- lapply(Ind.lst, function(x) {
  res <- filter_taxa(x, function(x) sum(x) >0, TRUE)
  return(res)
})

Venn_data <- list()
Venn_data <- lapply(Ind.lst, function(x) {
  names <- taxa_names(x)
  return(names)
})

names(Venn_data) <- c("100", "106", "107", "108", "112","113", "114", "115", "116", "117",
                      "121", "122", "123", "127", "128", "129", "132", "133", "135", "136",
                      "137", "138", "88", "89", "90", "92", "93", "95", "98", "99")

# Build a list for each host species

A_poly <- list(Venn_data[["88"]], Venn_data[["89"]], Venn_data[["90"]])
C_atri <- list(Venn_data[["92"]], Venn_data[["93"]], Venn_data[["95"]])
A_sexf <- list(Venn_data[["98"]], Venn_data[["99"]], Venn_data[["100"]])
A_whit <- list(Venn_data[["106"]], Venn_data[["107"]], Venn_data[["108"]])
P_molu <- list(Venn_data[["112"]], Venn_data[["113"]], Venn_data[["114"]])
P_ward <- list(Venn_data[["115"]], Venn_data[["116"]], Venn_data[["117"]])
S_apic <- list(Venn_data[["121"]], Venn_data[["122"]], Venn_data[["123"]])
S_nigr <- list(Venn_data[["127"]], Venn_data[["128"]], Venn_data[["129"]])
D_pseu <- list(Venn_data[["132"]], Venn_data[["133"]], Venn_data[["135"]])
D_pers <- list(Venn_data[["136"]], Venn_data[["137"]], Venn_data[["138"]])

#  Save data for import into http://bioinformatics.psb.urgent.be/webtools/Venn/

# save.excel(Venn_data, "Venn_data")

# Individual species plos
A_poly_plot <- venn.diagram(list(A=A_poly[[1]], B=A_poly[[2]],
                                 C=A_poly[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
C_atri_plot <- venn.diagram(list(A=C_atri[[1]], B=C_atri[[2]],
                                 C=C_atri[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
A_sexf_plot <- venn.diagram(list(A=A_sexf[[1]], B=A_sexf[[2]],
                                 C=A_sexf[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
A_whit_plot <- venn.diagram(list(A=A_whit[[1]], B=A_whit[[2]],
                                 C=A_whit[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
P_molu_plot <- venn.diagram(list(A=P_molu[[1]], B=P_molu[[2]],
                                 C=P_molu[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
P_ward_plot <- venn.diagram(list(A=P_ward[[1]], B=P_ward[[2]],
                                 C=P_ward[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
S_apic_plot <- venn.diagram(list(A=S_apic[[1]], B=S_apic[[2]],
                                 C=S_apic[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
S_nigr_plot <- venn.diagram(list(A=S_nigr[[1]], B=S_nigr[[2]],
                                 C=S_nigr[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
D_pseu_plot <- venn.diagram(list(A=D_pseu[[1]], B=D_pseu[[2]],
                                 C=D_pseu[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
D_pers_plot <- venn.diagram(list(A=D_pers[[1]], B=D_pers[[2]],
                                 C=D_pers[[3]]),
                            fill = rainbow(3), filename = NULL, cat.cex=0)
```

This section produces the Venn diagrams

```{r echo=TRUE, include=TRUE, tidy=TRUE}

plot_grid(gTree(children=A_poly_plot), gTree(children=A_sexf_plot), gTree(children=A_whit_plot), gTree(children=C_atri_plot), 
          gTree(children=P_molu_plot), gTree(children=D_pers_plot), gTree(children=D_pseu_plot), gTree(children=P_ward_plot), 
          gTree(children=S_apic_plot), gTree(children=S_nigr_plot),
          labels= c("A. polyacanthus", "A. sexfasciatus", "A. whitleyi", "C. atripectoralis", "P. moluccensis",
                    "D. perspicillatus", "D. pseudochrysopoecilus", "P. wardi", "S. apicalis",
                    "S. nigricans"), ncol=5, label_fontface = "italic", label_x=0, label_y=0.06, label_size=8)

# Save as a .pdf

ggsave("Venn_diagrams.pdf",
       plot_grid(gTree(children=A_poly_plot), gTree(children=A_sexf_plot), gTree(children=A_whit_plot), gTree(children=C_atri_plot), 
          gTree(children=P_molu_plot), gTree(children=D_pers_plot), gTree(children=D_pseu_plot), gTree(children=P_ward_plot), 
          gTree(children=S_apic_plot), gTree(children=S_nigr_plot),
          labels= c("A. polyacanthus", "A. sexfasciatus", "A. whitleyi", "C. atripectoralis", "P. moluccensis",
                    "D. perspicillatus", "D. pseudochrysopoecilus", "P. wardi", "S. apicalis",
                    "S. nigricans"), ncol=5, label_fontface = "italic", label_x=0, label_y=0.06, label_size=8),
       width=210, height=100, units="mm", dpi=600)
```

### Different feeding behaviours

Here we produce Venn diagrams for each of the feeding behaviours

```{r echo=TRUE, include=TRUE, tidy=TRUE}
diet.lst <- list()
for(i in levels(data.frame(sample_data(psI2rar))$Treatment)){
  tmp <- subset_samples(psI2rar, Treatment==i)
  diet.lst[[length(diet.lst)+1]] = tmp
}

diet.lst <- lapply(diet.lst, function(x) {
  res <- filter_taxa(x, function(x) sum(x) >0, TRUE)
  return(res)
})

Venn_data.diet <- list()
Venn_data.diet <- lapply(diet.lst, function(x) {
  names <- taxa_names(x)
  return(names)
})

names(Venn_data.diet) <- c("Planktivorous", "Territorial")

# Venn diagram for feeding behaviour

diet_venn_plot <-venn.diagram(list(A=Venn_data.diet[[1]], B=Venn_data.diet[[2]]),
                            fill = rainbow(2), filename = NULL, cat.cex=0)
```

## Different gut locations

And here we produce Venn diagrams for each of the gut locations

```{r echo=TRUE, include=TRUE, tidy=TRUE}
gut.lst <- list()
for(i in levels(data.frame(sample_data(ps))$Location)){
  tmp <- subset_samples(ps, Location==i)
  gut.lst[[length(gut.lst)+1]] = tmp
}

gut.lst <- lapply(gut.lst, function(x) {
  res <- filter_taxa(x, function(x) sum(x) >0, TRUE)
  return(res)
})

Venn_data.gut <- list()
Venn_data.gut <- lapply(gut.lst, function(x) {
  names <- taxa_names(x)
  return(names)
})

names(Venn_data.gut) <- c("1", "2", "3", "4")

# Venn diagram for feeding behaviour

gut_venn_plot <-venn.diagram(list(A=Venn_data.gut[[1]], B=Venn_data.gut[[2]], C=Venn_data.gut[[3]], D=Venn_data.gut[[4]]),
                            fill = rainbow(4), filename = NULL, cat.cex=0)
```

### Save Venn diagrams

Save the Venn diagrams as a single plot

```{r echo=TRUE, include=TRUE, tidy=TRUE}
plot_grid(gTree(children=diet_venn_plot), gTree(children=gut_venn_plot),
          labels= c("Feeding behaviour", "Gut location"), ncol=2, label_x=0, label_y=0.06, label_size=8)

ggsave("Venn_diagrams_feed_gut.pdf",
       plot_grid(gTree(children=diet_venn_plot), gTree(children=gut_venn_plot),
          labels= c("Feeding behaviour", "Gut location"), ncol=2, label_x=0, label_y=0.06, label_size=8),
       width=210, height=100, units="mm", dpi=600)
```

# Beta diversity 1 - Gut location

This section assess the role of gut location using GLMs and produces a barplot as output. Note, manyGLM analyses take 1+ hours to run.

Firslty, we agglomerate the data set to Class level as we are interested in which Classes of bacteria are changing throughout the gut

```{r echo=TRUE, include=TRUE, tidy=TRUE}
#Agglomerate to Class
ps.class <- tax_glom(ps, taxrank="Class")
```

mvabund is used here to test for differences in bacterial classes throughout the gut using a nested design

```{r echo=TRUE, include=TRUE, tidy=TRUE}
#Run mvabund model

class.otu <- data.frame(t(otu_table(ps.class)))
class.sample <- data.frame(sample_data(ps.class))

class.mv <- mvabund(class.otu)

class.int <- manyglm(class.mv ~ class.sample$FishSpecies*class.sample$Location,
                   family = "negative binomial")

class.spe <- manyglm(class.mv ~ class.sample$FishSpecies,
                   family = "negative binomial")

class.nested <- anova(class.spe, class.int, p.uni="adjusted", test="LR", nBoot=999, resamp="montecarlo")

```

ANOVA table sumamrising the results

```{r echo=TRUE, include=TRUE, tidy=TRUE}
class.nested$table
```

Summarise the results of the GLM looking for taxa which are significantly driving the results

```{r echo=TRUE, include=TRUE, tidy=TRUE}
p.values <- data.frame(class.nested$uni.p)
p.values <- data.frame(t(p.values[2,]))
colnames(p.values)[1] <- c("p_val")
test.stat <- data.frame(class.nested$uni.test)
test.stat <- data.frame(t(test.stat[2,]))
colnames(test.stat)[1] <- c("t_stat")
class_location.summary.stats <- cbind(test.stat, p.values)
row.names(class_location.summary.stats) <- sub('X', '', row.names(class_location.summary.stats))# Remove 'X' from OTU Ids

sig.location.class <- subset(class_location.summary.stats, p_val <= 0.05)[,c(1,2)]
```

## Classes that significantly change with location along the gut

```{r echo=TRUE, include=TRUE, tidy=TRUE}
sig.loc.class <- subset(otu_table(ps.class), rownames(otu_table(ps.class)) %in% row.names(sig.location.class))
sig.loc.class <- merge_phyloseq(sig.loc.class, tax_table(ps.class), sample_data(ps.class), phy_tree(ps.class),
                                refseq(ps.class))
tax_table(sig.loc.class)
otu_table(sig.loc.class)

class.loc.data <- psmelt(sig.loc.class)[,c(1,2,3,9,12,15:18)]

ASV_names <- list(
  "7a805ea2c6634094ebd78b4669369434"= "Saprospirae",
  "fe534f1229bb1b7a8ae1d289dee54966"= "Cytophagia",
  "133b140ce8485b3935ffc762e5007f90"= "Bacteroidia",
  "d8379007acfc27d6ddd8c99c7a664184"= "Flavobacteriia",
  "5647a07a4a526bb674a9792c65a7756c"= "Mollicutes",
  "e28e7ba09e89cf23875a9b4607658e2e"= "Erysipelotrichi",
  "3023adf7ffbf02e4ffac033d9f7381f3"= "Clostridia",
  "1bc2f1ba8b5778b613831487f858a248"= "Planctomycetia",
  "af806f6e83225057eb82f2bf35c6e77d"= "Verrucomicrobiae",
  "9a7283f9c2590a28af554fd89b10dba9"= "Deltaproteobacteria",
  "b727ff5efbd37a9e8b1675747fe4874e"= "Gammaproteobacteria"
)

ASV_labeller <- function(variable,value){
  return(ASV_names[value])
}


class.loc.box <- ggplot(class.loc.data, aes(Location, log(Abundance+1), color=Location)) +
  geom_boxplot() + theme_classic() +
  facet_wrap( ~ OTU , scales="free", labeller=ASV_labeller)+ 
  theme(axis.text.x = element_text(angle = 0, hjust = 1)) + theme(legend.position="none")

class.loc.box
ggsave("location_class.pdf", class.loc.box, width=297, height=210, units="mm")
```

# Beta diversit 2 - Feeding behaviour

Here we merge all samples for each indivudal and asses for differences in bacterial communities for each species of fish.

Firstly, ASVs with relative abundances <5% and not present in at least 3 samples are removed. The dataset eneds to be reduced to speed up computational time.

```{r echo=TRUE, include=TRUE, tidy=TRUE}
psIF = filter_taxa(psI, function(x) sum(x > 3) > (0.05*length(x)), TRUE)
```

## Nested design

```{r echo=TRUE, include=TRUE, tidy=TRUE}

#Run mvabund model
abund.otu <- data.frame(t(otu_table(psIF)))
ps.sample <- data.frame(sample_data(psIF))
treatment <- as.character(data.frame(sample_data(psIF))[,12])

ps.mv <- mvabund(abund.otu)


mod.int <- manyglm(ps.mv ~ ps.sample$FishSpecies*ps.sample$Treatment,
                   family = "negative binomial")

diet.nested <- anova(mod.int, p.uni="adjusted", test="LR", nBoot=999, resamp="montecarlo")

```

ANOVA table sumamrising the results

```{r echo=TRUE, include=TRUE, tidy=TRUE}
diet.nested$table
```

Results are summarised. We use a P values of < 0.10 to asses significance here to ensure we capture as many ASVsas possible

```{r echo=TRUE, include=TRUE, tidy=TRUE}
p.values <- data.frame(diet.nested$uni.p)
p.values <- data.frame(t(p.values[2,]))
colnames(p.values)[1] <- c("p_val")
test.stat <- data.frame(diet.nested$uni.test)
test.stat <- data.frame(t(test.stat[2,]))
colnames(test.stat)[1] <- c("t_stat")
summary.stats <- cbind(test.stat, p.values)
row.names(summary.stats) <- sub('X', '', row.names(summary.stats))# Remove 'X' from OTU Ids

sig.diet.sp <- subset(summary.stats, p_val <= 0.05)

```

## Feeding behaviour

ASVs that significantly change with the different feeding behaviour species

```{r echo=TRUE, include=TRUE, tidy=TRUE}
sig.diet.otus <- subset(otu_table(psIrar), rownames(otu_table(psIrar)) %in% row.names(sig.diet.sp))
sig.diet.otus <- merge_phyloseq(sig.diet.otus, tax_table(psIrar), sample_data(psIrar), phy_tree(psIrar),
                                refseq(psIrar))
```

Agglomerate to reduce the number of taxa

```{r echo=TRUE, include=TRUE, tidy=TRUE}
sig.diet.fam <- tax_glom(sig.diet.otus, taxrank="Family")
tax_table(sig.diet.fam)
otu_table(sig.diet.fam)
diet.data <- psmelt(sig.diet.fam)[,c(1,2,3,7,8,11,13, 15:19)]
```

## Plot results

This section produces two sets of boxplots representing significant ASV changes with host species and feeding behaviour

```{r echo=TRUE, include=TRUE, tidy=TRUE}

ASV_names <- list(
  "bec208ce410eba9fc44d1ae918fb7118"= "Cyclobacteriaceae",
  "bb4cd78ebed29bae22c4e0e6cb033123" = "Flammeovirgaceae",
  "f25abbe627a1c366346daba01df0edf1" = "Rikenellaceae",
  "0f8a32868d367f7f08d563a7e0b6a3b1" = "Cryomorphaceae",
  "9d252771a8d848fb05e4646c98295819" = "Flavobacteriaceae",
  "fdf0d4d0ff9c296a444eddf1f245393d" = "Erysipelotrichaceae",
  "3023adf7ffbf02e4ffac033d9f7381f3" = "Ruminococcaceae",
  "77cf7f7c86aba621c027e850038e2400" = "Lachnospiraceae",
  "1bc2f1ba8b5778b613831487f858a248" = "Pirellulaceae",
  "ae36d99bb59fbe002dd0ad4bbf3207b5" = "Puniceicoccaceae",
  "2ca3d710d77ff0b8d1697ce6d10a1465" = "koll13",
  "89bce984ee772ef456dcd0cd52af3e44" = "Desulfovibrionaceae",
  "886b568b1627d9fad5c2fb841fe62bbd" = "Desulfobulbaceae",
  "80249904dce392f3f1d75dec25929060" = "Desulfobacteraceae",
  "0f3fd5c10f86cba16811b85927559d49" = "Piscirickettsiaceae",
  "95ab37f66dcf6501b9a6258be99a7c79" = "HTCC2089",
  "1cd7f1533d1d561f5a9a37bc197ad4dd" = "Endozoicimonaceae",
  "5a8a8c32155609271ab41a52566c7017" = "Vibrionaceae",
  "b727ff5efbd37a9e8b1675747fe4874e" = "Pasteurellaceae",
  "b5d31c50e3ab8b7195c9ce3eb8bf3629" = "OM60")

diet.box <- ggplot(diet.data, aes(Treatment, log(Abundance+1), color=Treatment)) +
  geom_boxplot() + theme_classic() +
  facet_wrap( ~ OTU , scales="free", ncol=4, labeller=ASV_labeller)+ 
  theme(axis.text.x = element_text(angle = 0)) + theme(legend.position="none")

# Feeding behaviour boxplot
diet.box
```

Vibrio, Lachnospiracea and Pasteurellacea are the most different families in terms of rarefied abundance between the two treatment groups.

```{r echo=TRUE, include=TRUE, tidy=TRUE}

sig.diet.fam.sub <- subset_taxa(sig.diet.fam, Family=="Vibrionaceae" |Family== "Lachnospiraceae" |Family== "Pasteurellaceae")

diet.data <- psmelt(sig.diet.fam.sub)[,c(1,2,3,7,8,9,10, 15:19)]

ASV_names <- list(
  "77cf7f7c86aba621c027e850038e2400" = "Lachnospiraceae",
  "5a8a8c32155609271ab41a52566c7017" = "Vibrionaceae",
  "b727ff5efbd37a9e8b1675747fe4874e" = "Pasteurellaceae")

diet.box <- ggplot(diet.data, aes(Treatment, log(Abundance+1), color=Treatment)) +
  geom_boxplot() + theme_classic() +
  facet_wrap( ~ OTU , scales="free", ncol=1, labeller=ASV_labeller)+ 
  theme(axis.text.x = element_text(angle = 0)) + theme(legend.position="none")

# Feeding behaviour boxplot
diet.box

# Save output
ggsave("diet_boxplots.pdf", diet.box, width=210, height=297, units="mm")
```

# Boral - ordination plots

This analysis produces the ordinations plots for each phyla.

```{r echo=TRUE, include=TRUE, tidy=TRUE}
library(mvabund)
library(boral)
library(RColorBrewer)

psIF.rar <- rarefy_even_depth(psIF, sample.size = min(sample_sums(psIF)), rngseed = 1)

Proteo = subset_taxa(psIF.rar, Phylum=="Proteobacteria")
Bacter = subset_taxa(psIF.rar, Phylum=="Bacteroidetes")
Firmi = subset_taxa(psIF.rar, Phylum=="Firmicutes")
Planc = subset_taxa(psIF.rar, Phylum=="Planctomycetes")
Tener <- subset_taxa(psIF.rar, Phylum=="Tenericutes")
Spiro <-subset_taxa(psIF.rar, Phylum=="Spirochaetes")
Verru <- subset_taxa(psIF.rar, Phylum=="Verrucomicrobia")

Proteo = subset_taxa(psI, Phylum=="Proteobacteria")
y <- data.frame(t(otu_table(Proteo)))
fit.lvmnb1<-boral(y=y, family="negative.binomial", lv.control =list(num.lv=2), row.eff="fixed")

Bacter = subset_taxa(psI, Phylum=="Bacteroidetes")
x <- data.frame(t(otu_table(Bacter)))
fit.lvmnb2 <- boral(y=x, family="negative.binomial", lv.control =list(num.lv=2), row.eff="fixed")

Firmi = subset_taxa(psI, Phylum=="Firmicutes")
z <- data.frame(t(otu_table(Firmi)))
fit.lvmnb3 <- boral(y=z, family="negative.binomial", lv.control =list(num.lv=2), row.eff="fixed")

Planc = subset_taxa(psI, Phylum=="Planctomycetes")
w <- data.frame(t(otu_table(Planc)))
fit.lvmnb4 <- boral(y=w, family="negative.binomial", lv.control =list(num.lv=2), row.eff="fixed")

#alpha is an arguement in lvsplot(); relates to scaling
alpha <- 0.5 

testcov.1 <- fit.lvmnb1$lv.median %*% t(fit.lvmnb1$lv.coefs.median[, 2:3])
testcov.2 <- fit.lvmnb2$lv.median %*% t(fit.lvmnb2$lv.coefs.median[, 2:3])
testcov.3 <- fit.lvmnb3$lv.median %*% t(fit.lvmnb3$lv.coefs.median[, 2:3])
testcov.4 <- fit.lvmnb4$lv.median %*% t(fit.lvmnb4$lv.coefs.median[, 2:3])

#singular value decom
do.svd1 <- svd(testcov.1, fit.lvmnb1$num.lv, fit.lvmnb1$num.lv)
do.svd2 <- svd(testcov.2, fit.lvmnb2$num.lv, fit.lvmnb2$num.lv)
do.svd3 <- svd(testcov.3, fit.lvmnb3$num.lv, fit.lvmnb3$num.lv)
do.svd4 <- svd(testcov.4, fit.lvmnb4$num.lv, fit.lvmnb4$num.lv)

#svd has components d, u and v

#some math related to rescaling
choose.lvs1 <- scale(do.svd1$u * matrix(do.svd1$d[1:fit.lvmnb1$num.lv]^alpha, 
                                        nrow = fit.lvmnb1$n, ncol = 2, byrow = T), center = T, scale = F)
choose.lvs2 <- scale(do.svd2$u * matrix(do.svd2$d[1:fit.lvmnb2$num.lv]^alpha, 
                                        nrow = fit.lvmnb2$n, ncol = 2, byrow = T), center = T, scale = F)
choose.lvs3 <- scale(do.svd3$u * matrix(do.svd3$d[1:fit.lvmnb3$num.lv]^alpha, 
                                        nrow = fit.lvmnb3$n, ncol = 2, byrow = T), center = T, scale = F)
choose.lvs4 <- scale(do.svd4$u * matrix(do.svd4$d[1:fit.lvmnb4$num.lv]^alpha, 
                                        nrow = fit.lvmnb4$n, ncol = 2, byrow = T), center = T, scale = F)

treat <- sample_data(psIF.rar)$Treatment
speci <- sample_data(psIF.rar)$FishSpecies
speci <- as.factor(speci)

### Proteobacteria

choose.lvs1 <- as.data.frame(cbind(choose.lvs1, treat, speci))
choose.lvs1$treat <- as.factor(choose.lvs1$treat)
choose.lvs1$speci <- as.factor(choose.lvs1$speci)
colnames(choose.lvs1)[1:2] <- c("LV1", "LV2")

speci.hull.proteo <- NULL
for(i in levels(choose.lvs1$speci)){
  tmp <- choose.lvs1[choose.lvs1$speci == i, ][chull(choose.lvs1[choose.lvs1$speci == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(speci.hull.proteo)){speci.hull.proteo <- tmp} else 
    speci.hull.proteo <- rbind(speci.hull.proteo, tmp)
}

treat.hull.proteo <- NULL
for(i in levels(choose.lvs1$treat)){
  tmp <- choose.lvs1[choose.lvs1$treat == i, ][chull(choose.lvs1[choose.lvs1$treat == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(treat.hull.proteo)){treat.hull.proteo <- tmp} else 
    treat.hull.proteo <- rbind(treat.hull.proteo, tmp)
}

pa1 <- ggplot() +
  geom_polygon(data=speci.hull.proteo, aes(x=LV1, y=LV2, fill=speci, group=speci), alpha=0.3) +
  geom_point(data=choose.lvs1, aes(x=LV1, y=LV2, colour=speci), size=4) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))

pa2 <- ggplot() +
  geom_polygon(data=treat.hull.proteo, aes(x=LV1, y=LV2, fill=treat, group=treat), alpha=0.3) +
  geom_point(data=choose.lvs1, aes(x=LV1, y=LV2, colour=treat)) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.background = element_rect(colour="black", fill=NA, size=2))

### Bacteroidetes

choose.lvs2 <- as.data.frame(cbind(choose.lvs2, treat, speci))
choose.lvs2$treat <- as.factor(choose.lvs2$treat)
choose.lvs2$speci <- as.factor(choose.lvs2$speci)
colnames(choose.lvs2)[1:2] <- c("LV1", "LV2")

speci.hull.bacter <- NULL
for(i in levels(choose.lvs2$speci)){
  tmp <- choose.lvs2[choose.lvs2$speci == i, ][chull(choose.lvs2[choose.lvs2$speci == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(speci.hull.bacter)){speci.hull.bacter <- tmp} else 
    speci.hull.bacter <- rbind(speci.hull.bacter, tmp)
}

treat.hull.bacter <- NULL
for(i in levels(choose.lvs2$treat)){
  tmp <- choose.lvs2[choose.lvs2$treat == i, ][chull(choose.lvs2[choose.lvs2$treat == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(treat.hull.bacter)){treat.hull.bacter <- tmp} else 
    treat.hull.bacter <- rbind(treat.hull.bacter, tmp)
}

pb1 <- ggplot() +
  geom_polygon(data=speci.hull.bacter, aes(x=LV1, y=LV2, fill=speci, group=speci), alpha=0.3) +
  geom_point(data=choose.lvs2, aes(x=LV1, y=LV2, colour=speci), size=4) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))

pb2 <- ggplot() +
  geom_polygon(data=treat.hull.bacter, aes(x=LV1, y=LV2, fill=treat, group=treat), alpha=0.3) +
  geom_point(data=choose.lvs2, aes(x=LV1, y=LV2, colour=treat)) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))

### Firmicutes

choose.lvs3 <- as.data.frame(cbind(choose.lvs3, treat, speci))
choose.lvs3$treat <- as.factor(choose.lvs3$treat)
choose.lvs3$speci <- as.factor(choose.lvs3$speci)
colnames(choose.lvs3)[1:2] <- c("LV1", "LV2")

speci.hull.firmi <- NULL
for(i in levels(choose.lvs3$speci)){
  tmp <- choose.lvs3[choose.lvs3$speci == i, ][chull(choose.lvs3[choose.lvs3$speci == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(speci.hull.firmi)){speci.hull.firmi <- tmp} else 
    speci.hull.firmi <- rbind(speci.hull.firmi, tmp)
}

treat.hull.firmi <- NULL
for(i in levels(choose.lvs3$treat)){
  tmp <- choose.lvs3[choose.lvs3$treat == i, ][chull(choose.lvs3[choose.lvs3$treat == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(treat.hull.firmi)){treat.hull.firmi <- tmp} else 
    treat.hull.firmi <- rbind(treat.hull.firmi, tmp)
}

pc1 <- ggplot() +
  geom_polygon(data=speci.hull.firmi, aes(x=LV1, y=LV2, fill=speci, group=speci), alpha=0.3) +
  geom_point(data=choose.lvs3, aes(x=LV1, y=LV2, colour=speci), size=4) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))

pc2 <- ggplot() +
  geom_polygon(data=treat.hull.firmi, aes(x=LV1, y=LV2, fill=treat, group=treat), alpha=0.3) +
  geom_point(data=choose.lvs3, aes(x=LV1, y=LV2, colour=treat)) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))

### Planctomycetes

choose.lvs4 <- as.data.frame(cbind(choose.lvs4, treat, speci))
choose.lvs4$treat <- as.factor(choose.lvs4$treat)
choose.lvs4$speci <- as.factor(choose.lvs4$speci)
colnames(choose.lvs4)[1:2] <- c("LV1", "LV2")

speci.hull.planc <- NULL
for(i in levels(choose.lvs4$speci)){
  tmp <- choose.lvs4[choose.lvs4$speci == i, ][chull(choose.lvs4[choose.lvs4$speci == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(speci.hull.planc)){speci.hull.planc <- tmp} else 
    speci.hull.planc <- rbind(speci.hull.planc, tmp)
}

treat.hull.planc <- NULL
for(i in levels(choose.lvs4$treat)){
  tmp <- choose.lvs4[choose.lvs4$treat == i, ][chull(choose.lvs4[choose.lvs4$treat == i ,
                                                                 c("LV1", "LV2")]), ]
  if(is.null(treat.hull.planc)){treat.hull.planc <- tmp} else 
    treat.hull.planc <- rbind(treat.hull.planc, tmp)
}

pd1 <- ggplot() +
  geom_polygon(data=speci.hull.planc, aes(x=LV1, y=LV2, fill=speci, group=speci), alpha=0.3) +
  geom_point(data=choose.lvs4, aes(x=LV1, y=LV2, colour=speci), size=4) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))

pd2 <- ggplot() +
  geom_polygon(data=treat.hull.planc, aes(x=LV1, y=LV2, fill=treat, group=treat), alpha=0.3) +
  geom_point(data=choose.lvs4, aes(x=LV1, y=LV2, colour=treat)) +
  coord_equal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        panel.border = element_rect(colour="black", fill=NA))


ggsave("ordinations.pdf", 
       ggarrange(pa1, pa2, pb1, pb2, pc1, pc2, pd1, pd2, ncol=2), dpi=600, width=210, height=297, units="mm")
```
       
# Analysis of Chlorphyll

First we extract all Chloroplast and Cyanobacterial sequences and store as a phyloseq object 'Chloro'

```{r echo=TRUE, include=TRUE, tidy=TRUE}
Chloro = subset_taxa(data, Class=="Chloroplast" | Phylum=="Cyanobacteria")
Chloro
```

## Phylogenetic tree

Here we build a phylogenetic tree to look at differences between the two feeding behaviours

First build chloroplast data for each individual (merge gut locations)

```{r echo=TRUE, include=TRUE, tidy=TRUE}
ChloroI <- merge_samples(Chloro, "Individual")
sample_data(ChloroI)$Individual <- factor(sample_names(ChloroI))
ChloroI <- merge_phyloseq(ChloroI, seqs)
otu_table(ChloroI) <- t(otu_table(ChloroI))
sample_data(ChloroI) <- metadt
```

Build phyloseq object for the 20 most abuundant chloroplast ASVs

```{r echo=TRUE, include=TRUE, tidy=TRUE}
Chloro20 = names(sort(taxa_sums(ChloroI), TRUE)[1:20])
Chloro20 <- subset(otu_table(ChloroI), rownames(otu_table(ChloroI)) %in% Chloro20)
Chloro20 <- merge_phyloseq(Chloro20, tax_table(ChloroI), sample_data(ChloroI), phy_tree(ChloroI),
                       refseq(ChloroI))
tax_table(Chloro20)
otu_table(Chloro20)
sort(rowSums(otu_table(Chloro20)))
```

Build phylogenetic tree showing prevalance ofthe 20 most abundant chloroplast sequences among feeding behaviours
 
```{r echo=TRUE, include=TRUE, tidy=TRUE}
chloro_tree_plot <- plot_tree(Chloro20, color= "Treatment", label.tips="Order", ladderize="left", justify="left")
chloro_tree_plot 
ggsave("chloro_tree.pdf", chloro_tree_plot, width=297, height=210, units="mm")
```

## GLM model for species/diet difference

```{r echo=TRUE, include=TRUE, tidy=TRUE}

Chloro.otu <- data.frame(t(otu_table(ChloroI)))
Chloro.sample <- data.frame(sample_data(ChloroI))

Chloro.mv <- mvabund(Chloro.otu)

Chloro.mod <- manyglm(Chloro.mv ~ Chloro.sample$FishSpecies*Chloro.sample$Treatment, 
                       family = "negative binomial")
Chloro.spe<- manyglm(Chloro.mv ~ Chloro.sample$FishSpecies, 
                       family = "negative binomial")

Chloro.aov <- anova(Chloro.spe, Chloro.mod, p.uni="adjusted", test="LR", nBoot=999, resamp="montecarlo")

Chloro.aov$table
```

Analysis was notpowerful enough to detect Chloroplast ASVs that changed significantly with the model terms.
